### ISSUE TITLE
Merge `std` and `core`?

### ISSUE THREAD

**Issue by armanbilge** [armanbilge/oxidized#9]
The idea was to separate `cats-effect` core dependency, but in practice everything from fs2 up brings it in anyway. So it may be over-modularization.

There's an argument to just merge in kernel as well, who does concurrency without CE.

**Comment by armanbilge** [comment]
👍 to merging `kernel` and `std` since they both only need the CE kernel dependency, we can provide the `ConcurrentStateful` for `Ref` in a more convenient place.

I'm not sure why `core` independent of `kernel` is a big win. Unless you don't like `ConcurrentStateful` and/or it's not ready for primetime in which case I'm happy to punt that off :)

**Comment by bplommer** [comment]
> I'm not sure why core independent of kernel is a big win. 

Dunno, maybe it isn't 🤷

> Unless you don't like `ConcurrentStateful` and/or it's not ready for primetime in which case I'm happy to punt that off :)

Not that I don't like it but it's a lot more experimental than `core`. I like the idea of having a module that's just cats-mtl instances.

**Comment by armanbilge** [comment]
> it's a lot more experimental than `core`

Yes completely agree. Although anybody who cares about bincompatibility ~can~ should just copy-paste these instances wherever they want them :)

Separate modules doesn't really protect core from breaking bincompat in kernel, because it manifests as a breaking version bump for the whole repo anway. This is why the experimental CE CPS stuff is in its own repo with its own versioning.

Modules really only help with scoping dependencies I think.

**Comment by bpholt** [comment]
WDYT about moving the IO instances to a cats-mtl submodule (if they'll have it)? It'd be nice to have the `IOLocal`-based `Local` instance in a Typelevel repo if it's going to be more widespread due to something like Natchez or otel4s.

**Comment by armanbilge** [comment]
> WDYT about moving the IO instances to a cats-mtl submodule (if they'll have it)?

@bpholt yeah, the reason this project exists is because that's not possible 😅 CE testkit has a dependency on cats-mtl, so cats-mtl can't depend on CE without creating a cycle. Cycles again!

However, I am in favor of moving this somewhere more visible. And also generally reviving it :)

**Comment by bpholt** [comment]
Ugh, of course 😂😭

**Comment by armanbilge** [comment]
Btw, something that could be worth pursuing is moving `ConcurrentStateful` into MTL. It's just an interface, and it should really be the parent of `Stateful` I believe.

https://github.com/armanbilge/oxidized/blob/412be9cd0a60b901fd5f9157ea48bda8632c5527/kernel/src/main/scala/oxidized/ConcurrentStateful.scala#L19-L30
